src/IPM/IPM.jl

```diff
@@ -176,22 +178,29 @@ function InteriorPointSolver{KKTSystem}(nlp::AbstractNLPModel, opt::Options;
 
     aug_vec_length = is_reduced(kkt) ? n+m : n+m+nlb+nub
 
-    _w1 = is_reduced(kkt) ? ReducedKKTVector(n, m) : UnreducedKKTVector(n, m, nlb, nub)
-    _w2 = is_reduced(kkt) ? ReducedKKTVector(n, m) : UnreducedKKTVector(n, m, nlb, nub)
-    _w3 = is_reduced(kkt) ? ReducedKKTVector(n, m) : UnreducedKKTVector(n, m, nlb, nub)
-    _w4 = is_reduced(kkt) ? ReducedKKTVector(n, m) : UnreducedKKTVector(n, m, nlb, nub)
+    if is_reduced(kkt)
+        _w1 =  ReducedKKTVector(similar(x,n+m), n, m)
```

Comment:
Oups,  I think the syntax is a bit too complicated here. Should we create a new constructor for `ReducedKKTVector`? Something like
```julia
_w1 = ReducedKKTVector{T}(n, m)
```